test/grains_cases_test.rb

```diff
@@ -0,0 +1,87 @@
+require_relative 'test_helper'
+
+class GrainsTest < Minitest::Test
+  def test_test_name
+    test_case = GrainsCase.new(description: 'description')
+
+    assert_equal 'test_description', test_case.test_name
+  end
+
+  def test_test_name_when_description_has_spaces
+    test_case = GrainsCase.new(description: 'description with spaces')
+
+    assert_equal 'test_description_with_spaces', test_case.test_name
+  end
+
+  def test_workload_is_not_implemented
+    assert_raises(NotImplementedError) { GrainsCase.new.workload }
+  end
+
+  def test_skipped_with_zero_index
+    assert_equal '# skip', GrainsCase.new(index: 0).skipped
+  end
+
+  def test_skipped_with_nil_index
```

Comment:
It's the index of the set of test cases. Would something like this work:

```ruby
data = JSON.parse(data)
i = -1

cases = data['square']['cases'].map do |row|
  i += 1
  GrainsCase::SquareMethod.new(row.merge('index' => i))
end

cases << GrainsCase::TotalMethod.new(data['total'], index: i+1)
```

It adds a bit of complexity, but at least the index is correct.

Thank you so much for your help and feedback!